src/store/modules/nft.js

```diff
@@ -1,41 +1,271 @@
 /* eslint no-param-reassign: "off" */
 import Vue from 'vue';
-import { getNFTs } from '~/util/nft';
-import { getUserSellNFTClasses } from '@/util/api';
-
-const NFT_SET_USER_CLASSID_LIST_MAP = 'NFT_SET_USER_CLASSID_LIST_MAP';
+import * as api from '@/util/api';
+import {
+  NFT_INDEXER_LIMIT_MAX,
+  ORDER_CREATED_CLASS_ID_BY,
+  ORDER_COLLECTED_CLASS_ID_BY,
+  ORDER,
+  isWritingNFT,
+  isValidHttpUrl,
+  formatOwnerInfoFromChain,
+  formatNFTInfo,
+} from '~/util/nft';
+import * as TYPES from '../mutation-types';
 
 const state = () => ({
+  nftClassPurchaseInfo: {},
+  nftClassMetadata: {},
+  nftClassOwnerInfo: {},
   userClassIdListMap: {},
+  userLastCollectedTimestampMap: {},
 });
 
 const mutations = {
-  [NFT_SET_USER_CLASSID_LIST_MAP](state, { address, nfts }) {
+  [TYPES.NFT_SET_NFT_CLASS_PURCHASE_INFO](state, { classId, info }) {
+    Vue.set(state.nftClassPurchaseInfo, classId, info);
+  },
+  [TYPES.NFT_SET_NFT_CLASS_METADATA](state, { classId, metadata }) {
+    Vue.set(state.nftClassMetadata, classId, metadata);
+  },
+  [TYPES.NFT_SET_NFT_CLASS_OWNER_INFO](state, { classId, info }) {
+    Vue.set(state.nftClassOwnerInfo, classId, info);
+  },
+  [TYPES.NFT_SET_USER_CLASSID_LIST_MAP](state, { address, nfts }) {
     Vue.set(state.userClassIdListMap, address, nfts);
   },
+  [TYPES.NFT_SET_USER_LAST_COLLECTED_TIMESTAMP_MAP](
+    state,
+    { address, timestampMap }
+  ) {
+    Vue.set(state.userLastCollectedTimestampMap, address, timestampMap);
+  },
 };
 
+function compareIsWritingNFT(getters, classIdA, classIdB) {
+  const aIsWritingNFT = isWritingNFT(getters.getNFTClassMetadataById(classIdA));
+  const bIsWritingNFT = isWritingNFT(getters.getNFTClassMetadataById(classIdB));
+  if (aIsWritingNFT && !bIsWritingNFT) return -1;
+  if (!aIsWritingNFT && bIsWritingNFT) return 1;
+  return 0;
+}
+
 const getters = {
   NFTClassIdList: state => state.userClassIdListMap,
   getNFTClassIdListByAddress: state => address =>
     state.userClassIdListMap[address],
+  getNFTClassPurchaseInfoById: state => id => state.nftClassPurchaseInfo[id],
+  getNFTClassMetadataById: state => id => state.nftClassMetadata[id],
+  getNFTClassOwnerInfoById: state => id => state.nftClassOwnerInfo[id] || {},
+  getNFTClassOwnerOwnedNFTIds: (_, getters) => (id, address) =>
+    getters.getNFTClassOwnerInfoById(id)[address] || [],
+  getNFTClassOwnerOwnedNFTCount: (_, getters) => (id, address) =>
+    getters.getNFTClassOwnerOwnedNFTIds(id, address).length,
+  getNFTClassOwnerCount: state => id =>
+    Object.keys(state.nftClassOwnerInfo[id] || {}).length,
+  getNFTClassMintedCount: state => id =>
+    Object.values(state.nftClassOwnerInfo[id] || {}).reduce(
+      (acc, val) => acc + val.length,
+      0
+    ),
+  getUserLastCollectedTimestampByAddress: state => address =>
+    state.userLastCollectedTimestampMap[address],
+  getCreatedClassSorter: (_, getters) => (
+    classIds,
+    orderBy,
+    order = ORDER.DESC
+  ) => {
```

Comment:
I prefer using object for params, in case we add more in the future
```suggestion
  getCreatedClassIdSorter: (_, getters) => ({
    classIds,
    orderBy,
    order = ORDER.DESC
  }) => {
```